packages/gatsby/src/bootstrap/load-plugins/validate.ts

```diff
@@ -197,16 +197,23 @@ export async function validatePluginOptions({
           }
 
           try {
-            // All plugins have "plugins: []"" added to their options in load.ts, even if they
-            // do not have subplugins. We add plugins to the schema if it does not exist already
-            // to make sure they pass validation.
+            if (typeof plugin.pluginOptions === `undefined`) {
+              return null
+            }
+
             if (!optionsSchema.describe().keys.plugins) {
+              // All plugins have "plugins: []"" added to their options in load.ts, even if they
+              // do not have subplugins. We add plugins to the schema if it does not exist already
+              // to make sure they pass validation.
               optionsSchema = optionsSchema.append({
                 plugins: Joi.array().length(0),
               })
             }
 
-            await validateOptionsSchema(optionsSchema, plugin.pluginOptions)
+            plugin.pluginOptions = await validateOptionsSchema(
```

Comment:
I have no idea if this is the correct way to handle this @mxstbr, please feel free to laugh, then let me know how to do it correctly (or just take care of it).

Reply:
I have no idea either 😅 can we add a test somewhere that verifies this works correctly? :pray:

Reply:
Good call, also @wardpeet any insight here?

Reply:
We have an internal ticket for this :)
=> https://app.clubhouse.io/gatsbyjs/story/16679/override-plugin-options-with-defaults-defined-in-pluginoptionsschema

One thing that we need to flesh out before shipping this API without experimental is:
> How do we tell plugin authors that they have to make sure to have a fallback in place too. As this might give a false indication of safety